packages/core/content-manager/admin/src/history/components/VersionInputRenderer.tsx

```diff
@@ -0,0 +1,349 @@
+        type={props.attribute.customField}
+        disabled={fieldIsDisabled}
+      />
+    );
+  }
+
+  /**
+   * Since media fields use a custom input via the upload plugin provided by the useLibrary hook,
+   * we need to handle the them before other custom inputs coming from the useLibrary hook.
+   */
+  if (props.type === 'media') {
+    return <CustomMediaInput {...props} disabled={fieldIsDisabled} />;
+  }
+  /**
+   * This is where we handle ONLY the fields from the `useLibrary` hook.
+   */
+  const addedInputTypes = Object.keys(fields);
+  if (!attributeHasCustomFieldProperty(props.attribute) && addedInputTypes.includes(props.type)) {
+    const CustomInput = fields[props.type];
+    // @ts-expect-error – TODO: fix this type error in the useLibrary hook.
+    return <CustomInput {...props} hint={hint} disabled={fieldIsDisabled} />;
+  }
+
+  /**
+   * These include the content-manager specific fields, failing that we fall back
+   * to the more generic form input renderer.
+   */
+  switch (props.type) {
+    case 'blocks':
+      return <BlocksInput {...props} hint={hint} type={props.type} disabled={fieldIsDisabled} />;
+    case 'component':
+      return <ComponentInput {...props} hint={hint} disabled={fieldIsDisabled} />;
+    case 'dynamiczone':
+      return <DynamicZone {...props} hint={hint} disabled={fieldIsDisabled} />;
+    case 'relation':
+      return <CustomRelationInput {...props} hint={hint} disabled={fieldIsDisabled} />;
+    case 'richtext':
+      return <Wysiwyg {...props} hint={hint} type={props.type} disabled={fieldIsDisabled} />;
+    case 'uid':
+      return <UIDInput {...props} hint={hint} type={props.type} disabled={fieldIsDisabled} />;
+    /**
+     * Enumerations are a special case because they require options.
+     */
+    case 'enumeration':
+      return (
+        <FormInputRenderer
+          {...props}
+          hint={hint}
+          options={props.attribute.enum.map((value) => ({ value }))}
+          // @ts-expect-error – Temp workaround so we don't forget custom-fields don't work!
+          type={props.customField ? 'custom-field' : props.type}
+          disabled={fieldIsDisabled}
+        />
+      );
+    default:
+      // These props are not needed for the generic form input renderer.
+      const { unique: _unique, mainField: _mainField, ...restProps } = props;
+      return (
+        <FormInputRenderer
+          {...restProps}
+          hint={hint}
+          // @ts-expect-error – Temp workaround so we don't forget custom-fields don't work!
+          type={props.customField ? 'custom-field' : props.type}
+          disabled={fieldIsDisabled}
+        />
+      );
+  }
+};
+
+const attributeHasCustomFieldProperty = (
+  attribute: Schema.Attribute.AnyAttribute
+): attribute is Schema.Attribute.AnyAttribute & Schema.Attribute.CustomField<string> =>
+  'customField' in attribute && typeof attribute.customField === 'string';
+
+const useFieldHint = (
```

Comment:
🤔 Maybe I'll export this from InputRenderer instead of repeating it here

Reply:
great idea 👍